mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala

```diff
@@ -169,6 +181,33 @@ class LinearRegression(override val uid: String)
     val yMean = ySummarizer.mean(0)
     val yStd = math.sqrt(ySummarizer.variance(0))
 
```

Comment:
Currently `numFeatures` is obtained by summarization. In order to decide which algorithm is used for "auto" solver, it is necessary to get `numFeatures`. Is is ok to obtain `numFeatures` from first instance on dataset?